src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1407,11 +1407,23 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
                     }
                 }
                 ty::FnPtr(_) => {
+                    let unleash_miri = self
+                        .tcx
+                        .sess
+                        .opts
+                        .debugging_opts
+                        .unleash_the_miri_inside_of_you;
+                    let const_fn_ptr = self.tcx.features().const_fn_ptr;
                     if self.mode.requires_const_checking() {
-                        let mut err = self.tcx.sess.struct_span_err(
-                            self.span,
-                            &format!("function pointers are not allowed in const fn"));
-                        err.emit();
+                        if !(unleash_miri || const_fn_ptr) {
```

Comment:
Since the feature is inherently broken, maybe just do the unleash part and no feature gate?